src/server/.tests/ws-routes/controllers/Friends.spec.js

```diff
@@ -0,0 +1,482 @@
+			suite.friendsInstance.addNotification = sinon.stub()
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				socket.on(CLIENT_EVENTS.INVITATION_CONFIRMATION, data => {
+					return suite.friendsInstance.invitaitonConfirmation(data, socket, suite.connectionsMock)
+						.then(asserations)
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.INVITATION_CONFIRMATION, suite.REQUEST_MOCK)
+		
+			//then
+			function asserations() {
+				sinon.assert.calledOnce(suite.addFriendsStub)
+				done()
+			}
+		})
+
+		it('should add friends when invitation exists in database', done => {
+			//given
+			suite.userModelMock.find.returns(suite.queryResultMock)
+			suite.addFriendsStub = sinon.stub(suite.friendsInstance, 'addFriends').resolves()
+			suite.friendsInstance.addNotification = sinon.stub()
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				socket.on(CLIENT_EVENTS.INVITATION_CONFIRMATION, data => {
+					return suite.friendsInstance.invitaitonConfirmation(data, socket, suite.connectionsMock)
+						.then(asserations)
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.INVITATION_CONFIRMATION, suite.REQUEST_MOCK)
+
+			//then
+			function asserations() {
+				const expectedInvitatingUsername = suite.DUMMY_INVITATING_USERNAME
+				const expectedInvitatedUsername = suite.DUMMY_USERNAME
+				sinon.assert.calledWith(suite.addFriendsStub, expectedInvitatingUsername, expectedInvitatedUsername)
+				done()
+			}
+		})
+
+		it('should call sendMessageToSpecificUser and send confirmation message to invitataing user', done => {
+			//given
+			suite.userModelMock.find.returns(suite.queryResultMock)
+			suite.friendsInstance.addFriends = sinon.stub().resolves()
+			suite.utilsMock.addNotification.resolves()
+			suite.server.on(CLIENT_EVENTS.CONNECTION, socket => {
+				socket.on(CLIENT_EVENTS.INVITATION_CONFIRMATION, data => {
+					return suite.friendsInstance.invitaitonConfirmation(data, socket, suite.connectionsMock)
+						.then(() => asserations(socket))
+				})
+			})
+
+			//when
+			suite.client = socketClient.connect(SERVER_URL, SOCKET_OPTIONS)
+			suite.client.emit(CLIENT_EVENTS.INVITATION_CONFIRMATION, suite.REQUEST_MOCK)
+
+			//then
+			function asserations(socket) {
+				const expectedEventType = CLIENT_EVENTS.INVITATION_CONFIRMATION
+				const expectedRecievingUsername = suite.DUMMY_INVITATING_USERNAME
+				const expectedPayload = {username: suite.DUMMY_USERNAME}
+				sinon.assert.calledWith(suite.utilsMock.sendMessageToSepcificUser, socket, suite.connectionsMock, expectedRecievingUsername, expectedEventType, expectedPayload)
+				done()
+			}
+		})
+
+		it('should call addNotification from Notifications class to add new notification to pendingNotifications when recieving user is not connected to namespace', done => {
```

Comment:
Can we check instead if new notification has been pushed to `user.pendingNotifications` in found user? It would actually test it from business pov, instead of testing implementation.